homeassistant/components/mqtt/select.py

```diff
@@ -126,7 +126,7 @@ def message_received(msg):
             if value_template is not None:
                 payload = value_template.async_render_with_possible_json_value(payload)
 
-            if payload not in self.options:
+            if payload not in self.options and payload != "None":
```

Comment:
I agree that we should support a value to be the "unknown" value.

I looked into why it's `None` here and it's because zigbee2MQTT uses [`{{ value_json.${firstExpose.property} }}`](https://github.com/Koenkk/zigbee2mqtt/pull/7980/files#diff-5cd90a075b18a077fab2f7f5a4066f63eade9d49e5ef0b023bc3878272786e9eR511)

So this means that we can also decide if we want to have a different value for unknown. (maybe compare against lowercase `none`?) 

I also think that we should make it official that we mark it unknown by replacing payload with `None`. This PR currently relies on the logic in the select entity to mark it as unknown if it's not an option, however that's a safe guard. Probably better to prevent hitting that guard.

```python
if payload.lower() == "none":
    payload = None

self._attr_current_option = payload
```

Reply:
The reason I wrote it this way I did, that if "None" was a valid option in the list, it would work/become user selectable. But i guess that will still go south, so a guard will do